web/src/engine/websites/Zebrack.ts

```diff
@@ -0,0 +1,347 @@
+            data = await FetchProto<VolumeViewerViewV3>(request, protoTypes, 'Zebrack.VolumeViewerViewV3');
+        }
+        return data;
+    }
+
+    async fetchMagazineViewer(magazineId: string, magazineIssueId: string, secretKey: string): Promise<ZebrackResponse> {
+        const uri = new URL('/api/browser/magazine_viewer', this.apiURL);
+        uri.searchParams.set('secret', secretKey);
+        uri.searchParams.set('is_trial', '0');
+        uri.searchParams.set('os', 'browser');
+        uri.searchParams.set('magazine_id', magazineId);
+        uri.searchParams.set('magazine_issue_id', magazineIssueId);
+        let request = new FetchRequest(uri.href);
+        let data = await FetchProto<ZebrackResponse>(request, protoTypes, this.responseRootType);
+        if (!data.magazineViewerView) {
+            uri.searchParams.set('is_trial', '1');
+            request = new FetchRequest(uri.href);
+            data = await FetchProto<ZebrackResponse>(request, protoTypes, this.responseRootType);
+        }
+        return data;
+    }
+
+    async fetchGravureViewer(gravureId: string, secretKey: string): Promise<GravureViewerViewV3> {
+        const uri = new URL('/api/v3/gravure_viewer', this.apiURL);
+        uri.searchParams.set('secret', secretKey);
+        uri.searchParams.set('is_trial', '0');
+        uri.searchParams.set('os', 'browser');
+        uri.searchParams.set('gravure_id', gravureId);
+        let request = new FetchRequest(uri.href);
+        let data = await FetchProto<GravureViewerViewV3>(request, protoTypes, 'Zebrack.GravureViewerViewV3');
+        if (!data.images) {
+            uri.searchParams.set('is_trial', '1');
+            request = new FetchRequest(uri.href);
+            data = await FetchProto<GravureViewerViewV3>(request, protoTypes, 'Zebrack.GravureViewerViewV3');
+        }
+        return data;
+    }
+
+    async fetchChapterViewer(titleId: string, chapterId: string, secretKey: string): Promise<ChapterViewerViewV3> {
+        const uri = new URL('/api/v3/chapter_viewer', this.apiURL);
+        const params = new URLSearchParams();
+        params.set('secret', secretKey);
+        params.set('os', 'browser');
+        params.set('title_id', titleId);
+        params.set('chapter_id', chapterId);
+        params.set('type', 'normal');
+        const request = new FetchRequest(uri.href, {
+            method: 'POST',
+            body: params.toString(),
+            headers: {
+                'content-type': 'application/x-www-form-urlencoded'
+            }
+        });
+        return FetchProto<ChapterViewerViewV3>(request, protoTypes, 'Zebrack.ChapterViewerViewV3');
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const data = await Common.FetchImageAjax.call(this, page, priority, signal);
+        const key: string = page.Parameters ? page.Parameters['encryptionKey'] as string : undefined;
```

Comment:
I think the check for `page.Parameters` is redundant, this object is always defined.